controllers/metallb_controller_test.go

```diff
@@ -0,0 +1,56 @@
+package controllers
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/metallb/metallb-operator/api/v1alpha1"
+	"github.com/metallb/metallb-operator/test/consts"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("MetalLB Controller", func() {
+	Context("syncMetalLB", func() {
+		speakerImage := "test-speaker:latest"
+		controllerImage := "test-controller:latest"
+		BeforeEach(func() {
+			By("Setting the environment variables")
+			Expect(os.Setenv("SPEAKER_IMAGE", speakerImage)).To(Succeed())
```

Comment:
Testing these sounds like another test, maybe:
It("Should create manifests"
It("Should create manifests with image overrides

Reply:
It looks like we have no default for the images when the env vars are not set, so they need to be set anyway all the time.
Let's ignore the previous comment for now and go with one test